File: solutions/go/hamming/1/hamming.go
// Package hamming provides Hamming distance checks
package hamming

import "errors"

// Distance calculates the Hamming distance between two DNA strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 1, errors.New("Both strands must have the same length")
	}
	distance := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}
	return distance, nil
}

File: solutions/julia/hamming/1/hamming.jl
function distance(s1::AbstractString, s2::AbstractString)
    length(s1) == length(s2) ||
        throw(ArgumentError("Strands must have the same length"))
    dist = 0
    for (n1, n2) in zip(s1, s2)
        if n1 != n2
            dist += 1
        end
    end
    return dist
end
